src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -190,8 +190,10 @@ export default class SquarePaymentStrategy implements PaymentStrategy {
             throw new StandardError();
         }
 
-        if (errors && this._squareOptions && this._squareOptions.onError) {
-            this._squareOptions.onError(errors);
+        if (errors) {
+            if (this._squareOptions && this._squareOptions.onError) {
+                this._squareOptions.onError(errors);
+            }
             this._deferredRequestNonce.reject(errors);
```

Comment:
I would prefer to always reject it even if `errors` is empty, just to avoid a promise ended up being on hold. But this is still an improvement on what we currently have. Maybe cover this bugfix by adding a new unit test?

Reply:
hey @capsula4 , thanks for the feedback, I'm going to delegate that PR to you as I'm closing my laptop in few minutes.